src/brs/grpc/proto/BrsService.java

```diff
@@ -0,0 +1,106 @@
+package brs.grpc.proto;
+
+import brs.Blockchain;
+import brs.BlockchainProcessor;
+import brs.Generator;
+import brs.TransactionProcessor;
+import brs.grpc.GrpcApiHandler;
+import brs.grpc.handlers.*;
+import brs.services.AccountService;
+import brs.services.BlockService;
+import com.google.protobuf.Empty;
+import io.grpc.stub.StreamObserver;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class BrsService extends BrsApiServiceGrpc.BrsApiServiceImplBase {
+
+    private final Map<Class<? extends GrpcApiHandler<?,?>>, GrpcApiHandler<?,?>> handlers;
+
+    public BrsService(BlockchainProcessor blockchainProcessor, Blockchain blockchain, BlockService blockService, AccountService accountService, Generator generator, TransactionProcessor transactionProcessor) {
+        Map<Class<? extends GrpcApiHandler<?,?>>, GrpcApiHandler<?,?>> handlers = new HashMap<>();
+        handlers.put(GetMiningInfoHandler.class, new GetMiningInfoHandler(blockchainProcessor));
+        handlers.put(SubmitNonceHandler.class, new SubmitNonceHandler(blockchain, accountService, generator));
+        handlers.put(GetBlockHandler.class, new GetBlockHandler(blockchain, blockService));
+        handlers.put(GetAccountHandler.class, new GetAccountHandler(accountService));
+        handlers.put(GetAccountsHandler.class, new GetAccountsHandler(accountService));
+        handlers.put(GetTransactionHandler.class, new GetTransactionHandler(blockchain, transactionProcessor));
+        handlers.put(GetTransactionBytesHandler.class, new GetTransactionBytesHandler(blockchain, transactionProcessor));
+        this.handlers = Collections.unmodifiableMap(handlers);
+    }
+
+    private <T extends GrpcApiHandler<?,?>> T getHandler(Class<T> handlerClass) throws HandlerNotFoundException {
+        GrpcApiHandler<?, ?> handler = handlers.get(handlerClass);
+        if (!handlerClass.isInstance(handler)) {
+            throw new HandlerNotFoundException();
+        }
+        return handlerClass.cast(handler);
+    }
+
+    @Override
+    public void getMiningInfo(Empty request, StreamObserver<BrsApi.MiningInfo> responseObserver) {
+        try {
```

Comment:
a lot of ctrl-c, ctrl-v --- yes, it's straightforward code, but boilerplate... maybe, would be an idea to consider Lamda functions to make it more fancy :yum:

Reply:
I'll see what I can do 😄